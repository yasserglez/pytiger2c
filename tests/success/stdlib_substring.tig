let
    var a := "This is PyTiger2C"
in
    print(substring(a, 0, 0));
    print("\n");
    print(substring(a, 0, 1));
    print("\n");
    print(substring(a, 0, 2));
    print("\n");
    print(substring(a, 0, 3));
    print("\n");
    print(substring(a, 0, 4));
    print("\n");
    print(substring(a, 0, 5));
    print("\n");
    print(substring(a, 0, 6));
    print("\n");
    print(substring(a, 0, 7));
    print("\n");
    print(substring(a, 0, 8));
    print("\n");
    print(substring(a, 0, 9));
    print("\n");
    print(substring(a, 0, 10));
    print("\n");
    print(substring(a, 0, 11));
    print("\n");
    print(substring(a, 0, 12));
    print("\n");
    print(substring(a, 0, 13));
    print("\n");
    print(substring(a, 0, 14));
    print("\n");
    print(substring(a, 0, 15));
    print("\n");
    print(substring(a, 0, 16));
    print("\n");
    print(substring(a, 0, 17));
    print("\n");
    print(substring(a, 1, 0));
    print("\n");
    print(substring(a, 1, 1));
    print("\n");
    print(substring(a, 1, 2));
    print("\n");
    print(substring(a, 1, 3));
    print("\n");
    print(substring(a, 1, 4));
    print("\n");
    print(substring(a, 1, 5));
    print("\n");
    print(substring(a, 1, 6));
    print("\n");
    print(substring(a, 1, 7));
    print("\n");
    print(substring(a, 1, 8));
    print("\n");
    print(substring(a, 1, 9));
    print("\n");
    print(substring(a, 1, 10));
    print("\n");
    print(substring(a, 1, 11));
    print("\n");
    print(substring(a, 1, 12));
    print("\n");
    print(substring(a, 1, 13));
    print("\n");
    print(substring(a, 1, 14));
    print("\n");
    print(substring(a, 1, 15));
    print("\n");
    print(substring(a, 1, 16));
    print("\n");
    print(substring(a, 2, 0));
    print("\n");
    print(substring(a, 2, 1));
    print("\n");
    print(substring(a, 2, 2));
    print("\n");
    print(substring(a, 2, 3));
    print("\n");
    print(substring(a, 2, 4));
    print("\n");
    print(substring(a, 2, 5));
    print("\n");
    print(substring(a, 2, 6));
    print("\n");
    print(substring(a, 2, 7));
    print("\n");
    print(substring(a, 2, 8));
    print("\n");
    print(substring(a, 2, 9));
    print("\n");
    print(substring(a, 2, 10));
    print("\n");
    print(substring(a, 2, 11));
    print("\n");
    print(substring(a, 2, 12));
    print("\n");
    print(substring(a, 2, 13));
    print("\n");
    print(substring(a, 2, 14));
    print("\n");
    print(substring(a, 2, 15));
    print("\n");
    print(substring(a, 3, 0));
    print("\n");
    print(substring(a, 3, 1));
    print("\n");
    print(substring(a, 3, 2));
    print("\n");
    print(substring(a, 3, 3));
    print("\n");
    print(substring(a, 3, 4));
    print("\n");
    print(substring(a, 3, 5));
    print("\n");
    print(substring(a, 3, 6));
    print("\n");
    print(substring(a, 3, 7));
    print("\n");
    print(substring(a, 3, 8));
    print("\n");
    print(substring(a, 3, 9));
    print("\n");
    print(substring(a, 3, 10));
    print("\n");
    print(substring(a, 3, 11));
    print("\n");
    print(substring(a, 3, 12));
    print("\n");
    print(substring(a, 3, 13));
    print("\n");
    print(substring(a, 3, 14));
    print("\n");
    print(substring(a, 4, 0));
    print("\n");
    print(substring(a, 4, 1));
    print("\n");
    print(substring(a, 4, 2));
    print("\n");
    print(substring(a, 4, 3));
    print("\n");
    print(substring(a, 4, 4));
    print("\n");
    print(substring(a, 4, 5));
    print("\n");
    print(substring(a, 4, 6));
    print("\n");
    print(substring(a, 4, 7));
    print("\n");
    print(substring(a, 4, 8));
    print("\n");
    print(substring(a, 4, 9));
    print("\n");
    print(substring(a, 4, 10));
    print("\n");
    print(substring(a, 4, 11));
    print("\n");
    print(substring(a, 4, 12));
    print("\n");
    print(substring(a, 4, 13));
    print("\n");
    print(substring(a, 5, 0));
    print("\n");
    print(substring(a, 5, 1));
    print("\n");
    print(substring(a, 5, 2));
    print("\n");
    print(substring(a, 5, 3));
    print("\n");
    print(substring(a, 5, 4));
    print("\n");
    print(substring(a, 5, 5));
    print("\n");
    print(substring(a, 5, 6));
    print("\n");
    print(substring(a, 5, 7));
    print("\n");
    print(substring(a, 5, 8));
    print("\n");
    print(substring(a, 5, 9));
    print("\n");
    print(substring(a, 5, 10));
    print("\n");
    print(substring(a, 5, 11));
    print("\n");
    print(substring(a, 5, 12));
    print("\n");
    print(substring(a, 6, 0));
    print("\n");
    print(substring(a, 6, 1));
    print("\n");
    print(substring(a, 6, 2));
    print("\n");
    print(substring(a, 6, 3));
    print("\n");
    print(substring(a, 6, 4));
    print("\n");
    print(substring(a, 6, 5));
    print("\n");
    print(substring(a, 6, 6));
    print("\n");
    print(substring(a, 6, 7));
    print("\n");
    print(substring(a, 6, 8));
    print("\n");
    print(substring(a, 6, 9));
    print("\n");
    print(substring(a, 6, 10));
    print("\n");
    print(substring(a, 6, 11));
    print("\n");
    print(substring(a, 7, 0));
    print("\n");
    print(substring(a, 7, 1));
    print("\n");
    print(substring(a, 7, 2));
    print("\n");
    print(substring(a, 7, 3));
    print("\n");
    print(substring(a, 7, 4));
    print("\n");
    print(substring(a, 7, 5));
    print("\n");
    print(substring(a, 7, 6));
    print("\n");
    print(substring(a, 7, 7));
    print("\n");
    print(substring(a, 7, 8));
    print("\n");
    print(substring(a, 7, 9));
    print("\n");
    print(substring(a, 7, 10));
    print("\n");
    print(substring(a, 8, 0));
    print("\n");
    print(substring(a, 8, 1));
    print("\n");
    print(substring(a, 8, 2));
    print("\n");
    print(substring(a, 8, 3));
    print("\n");
    print(substring(a, 8, 4));
    print("\n");
    print(substring(a, 8, 5));
    print("\n");
    print(substring(a, 8, 6));
    print("\n");
    print(substring(a, 8, 7));
    print("\n");
    print(substring(a, 8, 8));
    print("\n");
    print(substring(a, 8, 9));
    print("\n");
    print(substring(a, 9, 0));
    print("\n");
    print(substring(a, 9, 1));
    print("\n");
    print(substring(a, 9, 2));
    print("\n");
    print(substring(a, 9, 3));
    print("\n");
    print(substring(a, 9, 4));
    print("\n");
    print(substring(a, 9, 5));
    print("\n");
    print(substring(a, 9, 6));
    print("\n");
    print(substring(a, 9, 7));
    print("\n");
    print(substring(a, 9, 8));
    print("\n");
    print(substring(a, 10, 0));
    print("\n");
    print(substring(a, 10, 1));
    print("\n");
    print(substring(a, 10, 2));
    print("\n");
    print(substring(a, 10, 3));
    print("\n");
    print(substring(a, 10, 4));
    print("\n");
    print(substring(a, 10, 5));
    print("\n");
    print(substring(a, 10, 6));
    print("\n");
    print(substring(a, 10, 7));
    print("\n");
    print(substring(a, 11, 0));
    print("\n");
    print(substring(a, 11, 1));
    print("\n");
    print(substring(a, 11, 2));
    print("\n");
    print(substring(a, 11, 3));
    print("\n");
    print(substring(a, 11, 4));
    print("\n");
    print(substring(a, 11, 5));
    print("\n");
    print(substring(a, 11, 6));
    print("\n");
    print(substring(a, 12, 0));
    print("\n");
    print(substring(a, 12, 1));
    print("\n");
    print(substring(a, 12, 2));
    print("\n");
    print(substring(a, 12, 3));
    print("\n");
    print(substring(a, 12, 4));
    print("\n");
    print(substring(a, 12, 5));
    print("\n");
    print(substring(a, 13, 0));
    print("\n");
    print(substring(a, 13, 1));
    print("\n");
    print(substring(a, 13, 2));
    print("\n");
    print(substring(a, 13, 3));
    print("\n");
    print(substring(a, 13, 4));
    print("\n");
    print(substring(a, 14, 0));
    print("\n");
    print(substring(a, 14, 1));
    print("\n");
    print(substring(a, 14, 2));
    print("\n");
    print(substring(a, 14, 3));
    print("\n");
    print(substring(a, 15, 0));
    print("\n");
    print(substring(a, 15, 1));
    print("\n");
    print(substring(a, 15, 2));
    print("\n");
    print(substring(a, 16, 0));
    print("\n");
    print(substring(a, 16, 1));
    print("\n");
    print(substring(a, 17, 0));
    print("\n");
    "end"
end
