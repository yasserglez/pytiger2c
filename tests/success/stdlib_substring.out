
T
Th
Thi
This
This 
This i
This is
This is 
This is P
This is Py
This is PyT
This is PyTi
This is PyTig
This is PyTige
This is PyTiger
This is PyTiger2
This is PyTiger2C

h
hi
his
his 
his i
his is
his is 
his is P
his is Py
his is PyT
his is PyTi
his is PyTig
his is PyTige
his is PyTiger
his is PyTiger2
his is PyTiger2C

i
is
is 
is i
is is
is is 
is is P
is is Py
is is PyT
is is PyTi
is is PyTig
is is PyTige
is is PyTiger
is is PyTiger2
is is PyTiger2C

s
s 
s i
s is
s is 
s is P
s is Py
s is PyT
s is PyTi
s is PyTig
s is PyTige
s is PyTiger
s is PyTiger2
s is PyTiger2C

 
 i
 is
 is 
 is P
 is Py
 is PyT
 is PyTi
 is PyTig
 is PyTige
 is PyTiger
 is PyTiger2
 is PyTiger2C

i
is
is 
is P
is Py
is PyT
is PyTi
is PyTig
is PyTige
is PyTiger
is PyTiger2
is PyTiger2C

s
s 
s P
s Py
s PyT
s PyTi
s PyTig
s PyTige
s PyTiger
s PyTiger2
s PyTiger2C

 
 P
 Py
 PyT
 PyTi
 PyTig
 PyTige
 PyTiger
 PyTiger2
 PyTiger2C

P
Py
PyT
PyTi
PyTig
PyTige
PyTiger
PyTiger2
PyTiger2C

y
yT
yTi
yTig
yTige
yTiger
yTiger2
yTiger2C

T
Ti
Tig
Tige
Tiger
Tiger2
Tiger2C

i
ig
ige
iger
iger2
iger2C

g
ge
ger
ger2
ger2C

e
er
er2
er2C

r
r2
r2C

2
2C

C

